test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py

```diff
@@ -148,6 +152,10 @@ def test_smmodelparallel_smdataparallel_mnist(instance_types, ecr_image, sagemak
     This test has been added for SM DataParallelism and ModelParallelism tests for re:invent.
     TODO: Consider reworking these tests after re:Invent releases are done
     """
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    if Version("1.9") <= Version(image_framework_version) < Version("1.10"):
```

Comment:
Please use a `SpecifierSet` instead of comparing to two versions.

```suggestion
    if Version(image_framework_version) in SpecifierSet(">=1.9,<1.10"):
```